src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        Node<K, V> newNode = new Node<>(hash, key, value, null);
 
+        if (table[hash] == null) {
+            table[hash] = newNode;
+        } else {
+            Node<K, V> current = table[hash];
+            while (current != null) {
+                if (current.hash == hash
+                        && (current.key == key || (key != null && key.equals(current.key)))) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = newNode;
+                    break;
+                }
+                current = current.next;
+            }
+        }
+
+        size++;
+        resizeIfNeeded();
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
```

Comment:
The same issue with the `hash` is in the `getValue` method. You should use the remainder of the division of the `hash` by the length of the `table` as the index.

Reply:
hash cannot be larger than table.length because the hashCode() method calculates a hash of the key, which is then used to look for the remainder of the division of 16